tests/datasets/test_geo.py

```diff
@@ -212,6 +221,29 @@ def sentinel(self, request: SubRequest) -> Sentinel2:
     def test_files(self, paths: Union[str, Iterable[str]]) -> None:
         assert 1 <= len(NAIP(paths).files) <= 2
 
+    def test_files_property_warns_non_existing_file(self, tmp_path: Path) -> None:
+        path = (tmp_path / "non_existing_file.tif").as_posix()
+        with pytest.warns(UserWarning, match="Path was ignored."):
+            MockRasterDataset(path).files
+
+    @pytest.mark.filterwarnings("ignore:.*Path was ignored.*")
```

Comment:
We don't want to ignore warnings, we want to ensure warnings: https://docs.pytest.org/en/7.4.x/how-to/capture-warnings.html#warns

Reply:
I see. the test above this ensures the warning. My idea was that these two tests different things. But I can merge them into one if that is better?

see `test_files_property_warns_non_existing_file`

Reply:
Let's merge them. They're both testing the same situation, we're just ensuring two things.

Reply:
Agreed